src/main/java/com/google/gwt/core/shared/GWT.java

```diff
@@ -62,17 +67,82 @@
 		 */
 		public Object create(Class<?> classLiteral);
 	}
-	
+
+    private static class SimplifiedGWTCreateSupplier implements CustomGWTCreateSupplier {
+
+        private  Map<Class<?>, Class<?>> implementations = new HashMap<>();
+
+        @Override
+        public Object create(Class<?> classLiteral) {
+            Class<?> implClass = implementations.get(classLiteral);
+            if (implClass != null) {
+                try {
+                    return implClass.newInstance();
+                } catch (InstantiationException | IllegalAccessException e) {
+                    throw new Error(e);
+                }
+            } else {
+                return null;
+            }
+        }
+        
+        public void register(Class<?> classLiteral, Class<?> classLiteralImplementation) {
+            implementations.put(classLiteral, classLiteralImplementation);
+        }
+    }
+	    
 	private static List<CustomGWTCreateSupplier> customSuppliers = Lists.newArrayList();
 	
+	private static SimplifiedGWTCreateSupplier simplifiedGWTCreateCustomSupplier;
+	
+	private static SimplifiedGWTCreateSupplier mockSupplier = createDefaultGWTCreateSupplier();
+	
 	public static void addCustomSupplier(CustomGWTCreateSupplier customSupplier) {
 		customSuppliers.add(customSupplier);
+		simplifiedGWTCreateCustomSupplier = null;
 	}
+
+    public static void registerGWTCreateImplementation(Class<?> classLiteral, Class<?> classLiteralImplementation) {
+        if (simplifiedGWTCreateCustomSupplier == null) {
```

Comment:
I fixed the filed set to null. It was intended to have single instance of "customSupplierClassToInstance" (new name) and reset it to null upon tearDown()

mockSupplier is non re-settable default  implementation provided by this Library.